File: Part8/C#/FunctionConcepts/FunctionConcepts_5.cs
// Print pizza cost with named parameters, in languages that allow this
using System;

class Program
{
    const double TOPPING_PRICE = 1.50;
    const double DELIVERY_FEE = 3.00;

    static void PrintPizzaCost ( string Size_Par, int ToppingCount_Par, bool DeliveryRequested_Par )
    {
        double basePrice = 0.0;
        
        if ( Size_Par == "small" )
        {
            basePrice = 12.99;
        }
        else if ( Size_Par == "medium" )
        {
            basePrice = 15.99;
        }
        else
        {
            basePrice = 18.99;
        }
        
        double toppingCost = ToppingCount_Par * TOPPING_PRICE;
        
        double deliveryFee = 0;
        if ( DeliveryRequested_Par == true )
        {
            deliveryFee = DELIVERY_FEE;
        }
        
        double totalPrice = basePrice + toppingCost + deliveryFee;
        
        Console.WriteLine ( "Pizza order:" );
        Console.WriteLine ( "\tBase price: {0:F2}", basePrice );
        Console.WriteLine ( "\tTopping price: {0:F2}", toppingCost );
        Console.WriteLine ( "\tDelivery price: {0:F2}", deliveryFee );
        Console.WriteLine ( "\t\tTotal price: {0:F2}", totalPrice );
    }

    static void Main ()
    {
        PrintPizzaCost ( Size_Par: "small", ToppingCount_Par: 2, DeliveryRequested_Par: true );
        
        string nextPizzaSize = "medium";
        int nextToppingCount = 1;
        bool nextPizzaIsDelivery = false;
        PrintPizzaCost ( Size_Par: nextPizzaSize, ToppingCount_Par: nextToppingCount, DeliveryRequested_Par: nextPizzaIsDelivery );
    }
}

File: Part8/Python/FunctionConcepts/FunctionConcepts_5.py
# Print pizza cost with named parameters, in languages that allow this
TOPPING_PRICE = 1.50
DELIVERY_FEE = 3.00

def printPizzaCost ( Size_Par, ToppingCount_Par, DeliveryRequested_Par ):
    basePrice = 0.0
    
    if Size_Par == "small":
        basePrice = 12.99
    elif Size_Par == "medium":
        basePrice = 15.99
    else:
        basePrice = 18.99
    
    toppingCost = ToppingCount_Par * TOPPING_PRICE
    
    deliveryFee = 0
    if DeliveryRequested_Par == True:
        deliveryFee = DELIVERY_FEE
    
    totalPrice = basePrice + toppingCost + deliveryFee
    
    print ( "Pizza order:" )
    print ( f"\tBase price: {basePrice:.2f}" )
    print ( f"\tTopping price: {toppingCost:.2f}" )
    print ( f"\tDelivery price: {deliveryFee:.2f}" )
    print ( f"\t\tTotal price: {totalPrice:.2f}" )

def main():
    printPizzaCost ( Size_Par="small", ToppingCount_Par=2, DeliveryRequested_Par=True )
    
    nextPizzaSize = "medium"
    nextToppingCount = 1
    nextPizzaIsDelivery = False
    printPizzaCost ( Size_Par=nextPizzaSize, ToppingCount_Par=nextToppingCount, DeliveryRequested_Par=nextPizzaIsDelivery )

if __name__ == "__main__":
    main()
